runtime/syntax/scad.yaml

```
@@ -0,0 +1,53 @@
+filetype: OpenSCAD
+
+# OpenSCAD is a functional programming language used for representing
+# 2D/3D models for use in the program of the same name.
+#
+# The following documents were used as reference material:
+# https://en.wikibooks.org/wiki/OpenSCAD_User_Manual/The_OpenSCAD_Language
+# https://openscad.org/cheatsheet/index.html
+
+detect:
+    filename: "\\.scad$"
+
+rules:
+    - identifier: "\\b(function|module) +[a-z0-9_]+"
+
+    - statement: "\\b(abs|acos|asin|assert|atan|atan2|ceil|child|children|chr|circle|color|concat|cos|cross|cube|cylinder|difference|dxf_cross|dxf_dim|each|echo|else|exp|floor|for|function|hull|if|import|import_dxf|intersection|intersection_for|is_bool|is_function|is_list|is_num|is_string|is_undef|len|let|linear_extrude|ln|log|lookup|max|min|minkowski|mirror|module|multmatrix|norm|offset|ord|parent_module|polygon|polyhedron|pow|projection|rands|render|resize|rotate|rotate_extrude|round|scale|search|sign|sin|sphere|sqrt|square|str|surface|tan|text|translate|union|version|version_num)\\b"
+
+    - symbol: "[,\\.;:?]"
+    - symbol.operator: "[-+*/%^<>!=]|[<=>!]=|&&|\\|\\|"
+    - symbol.brackets: "[{(<>)}]|\\[|\\]"
+
+    # modifiers that change interpretation of the subtree after it
+    - special: "[#%!*]"
+
+    # special variables start with a dollar sign
+    - special: "\\$[a-z]+\\b"
```

But this will highlight `$bar` in `foo$bar test`, am I right?
Since `\b` in front of `\$` doesn't work I suggest:
```suggestion
    - special: "(^|\\s+)\\$[a-z]+\\b"
```

But maybe I'm wrong?!
Sorry, but I didn't saw this at the first time.

---

OK, that makes sense.  I pushed another fix to include that, adjusted as follows.

Very tricky! However while x$y is a syntax error in openscad (not a valid special var),
it is valid to say x=$y.  For that reason, I used \B instead of the regex for start-of-line
or whitespace, in order to handle x=$y.  I struggled to come up with something and
finally realized that all I have to do is invert \b, since the simplest rule is just to say
that I don't want valid word characters in front of the $.  This also avoids a problem
with the whitespace rule, such that if somebody has some really weird color scheme
going on, like they change the background color or something, then the whitespace
won't show up as part of the special variable.

Thanks, I didn't even think about that, good to know someone reads these things.